src/main/kotlin/com/defenseunicorns/koci/Repository.kt

```diff
@@ -141,22 +141,40 @@ class Repository(
         }.status.isSuccess()
     }
 
+    suspend fun <T> fetch(descriptor: Descriptor, handler: (stream: InputStream) -> T): T {
+        return client.prepareGet(
+            when (descriptor.mediaType) {
+                MANIFEST_MEDIA_TYPE, INDEX_MEDIA_TYPE -> router.manifest(name, descriptor)
+                else -> router.blob(name, descriptor)
+            }
+        ) {
+            attributes.appendScopes(scopeRepository(name, ACTION_PULL))
+
+            when (descriptor.mediaType) {
+                MANIFEST_MEDIA_TYPE -> accept(ContentType.parse(MANIFEST_MEDIA_TYPE))
+                INDEX_MEDIA_TYPE -> accept(ContentType.parse(INDEX_MEDIA_TYPE))
+            }
+        }.execute { res ->
```

Comment:
Add error handling?

Reply:
This is a generic function, so leaving error handling up to the caller.

Reply:
Ok so non 200s are being handled by the callers now? I guess by wrapping everything in runCatching it will handle it gracefully just might be hard to debug non 200 related failures?